std/node/_process.ts

```diff
@@ -0,0 +1,167 @@
+// Copyright 2018-2020 the Deno authors. All rights reserved. MIT license.
+import { notImplemented } from "./_utils.ts";
+
+/** https://nodejs.org/api/process.html#process_process_arch */
+export const arch = Deno.build.arch;
+
+/** https://nodejs.org/api/process.html#process_process_chdir_directory */
+export const chdir = Deno.chdir;
+
+/** https://nodejs.org/api/process.html#process_process_cwd */
+export const cwd = Deno.cwd;
+
+/** https://nodejs.org/api/process.html#process_process_exit_code */
+export const exit = Deno.exit;
+
+/** https://nodejs.org/api/process.html#process_process_pid */
+export const pid = Deno.pid;
+
+/** https://nodejs.org/api/process.html#process_process_platform */
+export const platform = Deno.build.os === "windows" ? "win32" : Deno.build.os;
+
+/** https://nodejs.org/api/process.html#process_process_version */
+export const version = `v${Deno.version.deno}`;
+
+/** https://nodejs.org/api/process.html#process_process_versions */
+export const versions = {
+  node: Deno.version.deno,
+  ...Deno.version,
+};
+
+/** https://nodejs.org/api/process.html#process_process_nexttick_callback_args */
+function nextTick(this: unknown, cb: () => void): void;
+function nextTick<T extends Array<unknown>>(
+  this: unknown,
+  cb: (...args: T) => void,
+  ...args: T
+): void;
+function nextTick<T extends Array<unknown>>(
+  this: unknown,
+  cb: (...args: T) => void,
+  ...args: T
+) {
+  if (args) {
+    queueMicrotask(() => cb.call(this, ...args));
+  } else {
+    queueMicrotask(cb);
+  }
+}
+
+/** https://nodejs.org/api/process.html#process_process */
+// @deprecated `import { process } from 'process'` for backwards compatibility with old deno versions
+export const process = {
+  /** https://nodejs.org/api/process.html#process_process_arch */
+  arch: Deno.build.arch,
+  /** https://nodejs.org/api/process.html#process_process_argv */
+  get argv(): string[] {
+    // Getter delegates --allow-env and --allow-read until request
+    // Getter also allows the export Proxy instance to function as intended
+    return [Deno.execPath(), ...Deno.args];
+  },
+  /** https://nodejs.org/api/process.html#process_process_chdir_directory */
+  chdir: Deno.chdir,
+  /** https://nodejs.org/api/process.html#process_process_cwd */
+  cwd: Deno.cwd,
+  /** https://nodejs.org/api/process.html#process_process_exit_code */
+  exit: Deno.exit,
+  /** https://nodejs.org/api/process.html#process_process_env */
+  get env(): { [index: string]: string } {
+    // Getter delegates --allow-env and --allow-read until request
+    // Getter also allows the export Proxy instance to function as intended
+    return Deno.env.toObject();
+  },
```

Comment:
If the problem is that this getter doesn't support the named exports destructuring, wouldn't the better solution to be a Proxy as you say to rather get the benefit for both cases?

Reply:
Proxy should be the solution, however a proxy-ed variable still seems to be triggering read/env operations

Reply:
How are you seeing that? The proxy should be based on an empty object base that only populates on the handler calls. I believe it's possible.

Reply:
Perhaps, but I don't know how that would work on a named import :/

Reply:
It should be as simple as `env: new Proxy(...)` right?

Reply:
I don't understand what you mean by that

Reply:
Something like the following:

```suggestion
  env: new Proxy(proxy_base, {
    set () {
      initProxy(envBase);
      return Reflect.set(...arguments);
    },
    ownKeys () {
      initProxy(envBase);
      return Reflect.ownKeys(envBase);
    },
    ... for defineProperty, deleteProperty, get, getOwnPropertyDescriptor, 
  }),
```

Reply:
Where the `initEnv` function does `Object.assign(envBase, Deno.env.toObject())` sort of thing.

Reply:
@guybedford After practicing a little with Proxies I think it's safe to say that perhaps the module separation might be the best choice. Even though proxying the properties of the env variables works to prevent an actual permission operation, the real problem comes when you try to read the env property as a whole. Since `envBase` is an empty object that doesn't have the actual env information until it's requested, logging it as a whole property will show just an empty object, deviating from the Node implementation.

Reply:
@Soremwar inspection should result in the `ownKeys` hook as above explicitly providing the full object. The output of console.log is hardly the normative comparison here. Even then it could be made to work with a special log symbol. `JSON.stringify` etc will work fine too.

Reply:
@guybedford I tried that with the `ownKeys` hook, but it doesn't seem to be triggered when you inspect the variable. Am I missing something?

Reply:
Inspection and usage are completely different things. Equality is not based on inspection, and inspection code paths can be adjusted using eg special symbols or any time of local inspection hook to adjust the inspection output.

Otherwise we'd never be able to use proxies at all, just because they log different doesn't mean we can't use them as replacements for other objects.

Reply:
How would I go about implementing this custom log hook then? Sorry, I'm a little inexperienced in this area

I already have the proxies for argv and env so that would be the last discrepancy that needs to be taken care of

Reply:
@Soremwar the inspection code can be found at https://github.com/denoland/deno/blob/master/runtime/js/02_console.js#L400, where you can follow the proxy path as well. For example, a private `envBase[internal.logSymbol] = function () { return 'custom logger' }` where the `internal.logSymbol` is only accessible to Deno core code only is one approach. There are many ways to do it so you'd have to check the exact approach with another reviewer on that.

Reply:
Yep, I found the function to custom inspect the variables, so it's all taken care off...sorta

As it turns out the inspect function will swallows any errors so I can't rely on it to test the permissions until #9095 gets merged

Reply:
Nevermind, I ended up solving an unrelated bug, this was totally doable right now